server/spec/suspend_mode_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'spec_helper'
+require 'candlepin_scenarios'
+
+describe 'Suspend mode' do
+  include CandlepinMethods
+
+  after(:each) do
+    resetMode
+  end
+
+  def resetMode
+    @cp.post('/status/mode?mode=NORMAL&reason=Mode test reset')
+  end
+
+  def changeMode(mode)
+    @cp.post("/status/mode?mode=#{mode}&reason=test")
+  end
+
+  it 'should provide /status/mode endpoint' do
+    mode = @cp.get('/status/mode')
+    expect(mode).to have_key("mode")
+    expect(mode["mode"]).to eq("NORMAL")
+    changeMode("SUSPEND")
+
+    mode = @cp.get('/status/mode')
+    expect(mode).to have_key("mode")
+    expect(mode["mode"]).to eq("SUSPEND")
+  end
+
+  it 'should not accept requests when in SUSPEND mode' do
+    changeMode("SUSPEND")
+    expect { create_owner random_string('test_owner') }.to raise_error(RestClient::ServiceUnavailable)
```

Comment:
Here we could test that returned error is machine-readable as per the design doc